src/resolve/treeContainers.ts

```diff
@@ -166,9 +167,23 @@ export class ZipTreeContainer extends TreeContainer {
     throw new LibraryError('error_expected_file_path', fsPath);
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   public readFileSync(fsPath: string): Buffer {
-    throw new Error('Method not implemented');
+    let done = false;
+    let dataBuffer: Buffer;
+    this.readFile(fsPath)
+      .then((buffer) => {
+        dataBuffer = buffer;
+        done = true;
+      })
+      .catch((error) => {
+        done = true;
+        throw error;
+      });
+
```

Comment:
This module covers doing exactly the ask here.  Do something async but treat the current flow and synchronous.